spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -1,62 +1,144 @@
-#         uid: '123545',
-#         info: {
-#           email: 'test@example.com'
-#         }
-#       )
-#     end
-
-#     before do
-#       OmniAuth.config.test_mode = true
-#       OmniAuth.config.mock_auth[:openid_connect] = auth
-#       request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:openid_connect]
-#       request.env['devise.mapping'] = Devise.mappings[:user] # If using Devise
-#     end
-
-#     let(:user) { create(:user) } # Defining the user
-
-#     context 'when the email is missing and user does not exist' do
-#       before do
-#         allow(User).to receive(:from_omniauth).and_return(nil)
-#         allow(auth.info).to receive(:email).and_return(nil)
-#         get :openid_connect
-#       end
-
-#       it 'redirects to the registration page with a flash message' do
-#         expect(flash[:notice]).to eq('Something went wrong, Please try signing-up here.')
-#         expect(response).to redirect_to(new_user_registration_path)
-#       end
-#     end
-
-#     context 'with correct credentials' do
-#       before do
-#         create(:org, managed: false, is_other: true)
-#         @org = create(:org, managed: true)
-#         @identifier_scheme = create(:identifier_scheme,
-#                                     name: 'openid_connect',
-#                                     description: 'CILogon',
-#                                     active: true,
-#                                     identifier_prefix: 'https://www.cilogon.org/')
-
-#         Rails.application.env_config['devise.mapping'] = Devise.mappings[:user]
-#         Rails.application.env_config['omniauth.auth'] = OmniAuth.config.mock_auth[:openid_connect]
-#         allow(User).to receive(:from_omniauth).and_return(user)
-#         # get :openid_connect
-#       end
-
-#       it 'links account from external credentials' do
-#         expect(flash[:notice]).to eq('Linked successfully')
-#         expect(response).to redirect_to(root_path)
-#       end
-#     end
-#   end
-# end
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+  before do
+    # Enable test mode for OmniAuth
+    OmniAuth.config.test_mode = true
+
+    # Setup Devise mapping
+    @request.env["devise.mapping"] = Devise.mappings[:user]
+    create(:org, managed: false, is_other: true)
+    @org = create(:org, managed: true)
+    @identifier_scheme = create(:identifier_scheme,
+                                name: 'openid_connect',
+                                description: 'CILogon',
+                                active: true,
+                                identifier_prefix: 'https://www.cilogon.org/')
+
+    # Mock OmniAuth data for OpenID Connect with necessary info
+    OmniAuth.config.mock_auth[:openid_connect] = OmniAuth::AuthHash.new({
+      provider: 'openid_connect',
+      uid: '12345',
+      info: {
+        email: 'test@example.com',
+        first_name: 'Test',
+        last_name: 'User',
+        name: 'Test User'
+      }
+    })
```

Comment:
We don't need to define this here since it is already on `spec_helper.rb`

https://github.com/portagenetwork/roadmap/blob/4da5fddbea10a0f8d907083a2462cea941b29325/spec/spec_helper.rb#L126-L136